Best practices/Security guidelines.md

```diff
@@ -32,132 +23,111 @@ services.AddCors(options =>
-                context.Response.Headers.Add("X-Content-Type-Options", "nosniff");
-                context.Response.Headers.Add("X-Xss-Protection", "1");
-                context.Response.Headers[HeaderNames.CacheControl] = "no-cache, no-store";
-                await next().ConfigureAwait(false);
-            });
-```
+There are a couple of attacks that can be prevented by adding a few security headers. Here are some that we recommend using:
 
-To restrict the application to only be accessible by HTTPS calls you can use the following sample of code:
+- [`X-Frame-Options`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options) - indicates whether or not a browser should be allowed to render a page in a frame object, which is used to avoid click-jacking attacks by ensuring that their content is not embedded into other sites.
+- [`X-Content-Type-Options`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Content-Type-Options) - indicates that the MIME types advertised in the `Content-Type` headers should be followed, which is used to avoid MIME-type sniffing.
+- [`X-XSS-Protection`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-XSS-Protection) - a feature of some browsers that stops pages from loading when they detect reflected cross-site scripting (XSS) attacks. Note that this header is non-standard and is not on a standards track, meaning that it will not work for every user. This is also largely unnecessary in modern browsers with implementations of content security policies.
+- [`Cache-Control`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control) - controls caching in browsers and shared caches. By setting the value to `no-cache`, we instruct the cache to always validate the content with the origin server before returning it to the requester, which ensures that the requester will always get the newest version of the content.
 
-**Startup.cs**:
+We can add these headers to all our responses by adding a middleware that will add the headers to every response:
 
 ```c#
-app.UseHsts();
-app.UseHttpsRedirection();
+app.Use(async (context, next) =>
+    {
+        context.Response.Headers.Add("X-Frame-Options", "deny");
+        context.Response.Headers.Add("X-Content-Type-Options", "nosniff");
+        context.Response.Headers.Add("X-Xss-Protection", "1");
+        context.Response.Headers[HeaderNames.CacheControl] = "no-cache, no-store";
+        await next().ConfigureAwait(false);
+    });
 ```
 
-
+When using these headers, we must have in mind the fact that they are only instructions for the browsers. This means that we're depending on the "goodwill" of browser developers to implement the mechanisms we advise using. Since most of our users will use one of the more popular browsers which do support these headers, this will not be an issue for them (and us), but these headers are not do not guarantee protection like other security guidelines which are implemented on the server side.
```

Comment:
`but these headers are not do not guarantee protection like other security guidelines` - we should fix the `are not do not` part in this sentence

Reply:
I will never stop being amazed by how many times I can read my own text and not notice obvious mistakes like these 😄 oh well, that's why have reviews :)